### PR TITLE
Add manual game start capability to room host

### LINKED ISSUE #236: Better ready/unready players information
We previously attempted to add manual start for the room host on #238, but as 
@indocomsoft pointed out the manual state changing caused some dangerous side effects in room  state changes. 

As an alternative, we can just display better information about ready/unready players in the room `PREPARED` state, so you know who to pester to ready up.

Possible extension: host to kick people from the room who may be coming in from a buggy device?

### PR DESCRIPTION
**Pull request by juxd**
Fix #236

### CODE REVIEW

**Comment by indocomsoft** [comment]
I think this would not work -- the previous state machine requires all the clients to have set their `result` to `null`. Otherwise, if the host force start the game when no one else has marked themselves as ready, the game will immediately end because the game will automatically transition to `COMPLETED`

File: Web/src/components/create-room/HostRoom.tsx [hunk 1]
```diff
@@ -220,10 +226,88 @@ const HostRoomLobby: React.FC<FinishedComponentProps> = ({
...
+    && css`
+      @media (max-width: ${mdMin}) {
+        display: none;
+      }
+  `};
+`;
+
+const InverseButton = styled(BigButton)`
+  background: white;
+  color: var(--purple);
```
Comment (hzxcaryn) [inline Web/src/components/create-room/HostRoom.tsx]:
```suggestion
  color: var(--dark-purple);
```

File: Web/src/components/create-room/HostRoom.tsx [hunk 2]
```diff
@@ -220,10 +226,88 @@ const HostRoomLobby: React.FC<FinishedComponentProps> = ({
...
-  />
-);
+type ContainerProps = {
+  hideOnMedium?: boolean;
+  hideOnLarge?: boolean;
+};
+
+const NumPlayersContainer = styled.div`
+  display: flex;
+  justify-content: flex-start;
```
Comment (hzxcaryn) [inline Web/src/components/create-room/HostRoom.tsx]: Actually hor these does nothing
```suggestion
```

File: Web/src/components/create-room/HostRoom.tsx [hunk 3]
```diff
@@ -220,10 +226,88 @@ const HostRoomLobby: React.FC<FinishedComponentProps> = ({
...
+    margin: 0 0.25rem;
+  }
+
+  && > svg {
+    height: 2em;
+  }
+
+  @media (max-width: ${mdMin}) {
+    justify-content: center;
+  }
```
Comment (hzxcaryn) [inline Web/src/components/create-room/HostRoom.tsx]: same as above cos justify-content does nothing here

File: Web/src/components/create-room/HostRoom.tsx [hunk 4]
```diff
@@ -220,10 +226,88 @@ const HostRoomLobby: React.FC<FinishedComponentProps> = ({
...
+        display: flex;
+      }
+  `};
+
+  ${(props: ContainerProps) => props.hideOnMedium
+    && css`
+      @media (max-width: ${mdMin}) {
+        display: none;
+      }
+  `};
```
Comment (hzxcaryn) [inline Web/src/components/create-room/HostRoom.tsx]: Haha don't need these also for this use case

**Comment by juxd** [comment]
lol looks like we're closing this in favour of just showing ready/unready players? can we get a confirmation @indocomsoft

**Comment by indocomsoft** [comment]
Yeah, I think it's better to just show the ready/unready players so those who are ready can poke those who are not ready yet

**Pull request closed without merging**